### PR TITLE
feat: update `@ui5/webcomponents` to `1.17.0`

### LINKED ISSUE #4920: [Select]: Icon not rendering as child of Option
### Describe the bug

I want to create a custom color picker using the `Select` component that renders the color as a icon and the color name as a label. Even though the documentation says that the `Option` subcomponent of the `Select` component renders arbitrary ReactNodes, `Icon` components do not get rendered when added as children. 

I saw that the `Option` component itself has an `icon` property. However, there is now option to style this icon and I need to change the icons color for each option.

### Isolated Example

https://codesandbox.io/p/sandbox/festive-lederberg-zr7rwn?file=%2Fsrc%2FApp.tsx%3A41%2C10

### Reproduction steps

See sandbox.

### Expected Behaviour

I would expect the Option to render the Icon element as a child, next to the label. It would also work if there is a simple way to style the icon that is passed as property to the `Option` component.

### Screenshots or Videos

_No response_

### UI5 Web Components for React Version

1.17.0

### UI5 Web Components Version

1.15.0

### Browser

Chrome

### Operating System

_No response_

### Additional Context

_No response_

### Relevant log output

_No response_

### Declaration

- [X] I’m not disclosing any internal or sensitive information.

### PR DESCRIPTION
**Pull request by Lukas742**
Closes #4920

### CODE REVIEW

File: packages/main/src/webComponents/SelectMenuOption/index.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+   */
+  displayText?: string;
+  /**
+   * **Note:** The property is inherited and not supported. If set, it won't take any effect.
+   */
+  navigated?: boolean;
+  /**
+   * **Note:** The property is inherited and not supported. If set, it won't take any effect.
+   */
+  type?: ListItemType | keyof typeof ListItemType;
```
Comment (MarcusNotheis) [inline packages/main/src/webComponents/SelectMenuOption/index.tsx:21]: we should remove these, just like for the segmented button

File: packages/main/src/webComponents/SelectMenuOption/index.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+   * Defines the value of the `Select` inside an HTML Form element when this component is selected. For more information on HTML Form support, see the `name` property of `Select`.
+   */
+  value?: string;
+}
+
+export interface SelectMenuOptionDomRef extends SelectMenuOptionAttributes, Ui5DomRef {
+  /**
+   * **Note:** The property is inherited and not supported. If set, it won't take any effect.
+   */
+  accessibilityAttributes: Record<string, unknown>;
```
Comment (MarcusNotheis) [inline packages/main/src/webComponents/SelectMenuOption/index.tsx:32]: remove

File: packages/main/src/webComponents/SelectMenuOption/index.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  /**
+   * **Note:** The slot is inherited and not supported. If set, it won't take any effect.
+   *
+   * __Note:__ This prop will be rendered as [slot](https://www.w3schools.com/tags/tag_slot.asp) (`slot="deleteButton"`).
+   * Since you can't change the DOM order of slots when declaring them within a prop, it might prove beneficial to manually mount them as part of the component's children, especially when facing problems with the reading order of screen readers.
+   *
+   * __Note:__ When passing a custom React component to this prop, you have to make sure your component reads the `slot` prop and appends it to the most outer element of your component.
+   * Learn more about it [here](https://sap.github.io/ui5-webcomponents-react/?path=/docs/knowledge-base-handling-slots--docs).
+   */
+  deleteButton?: unknown;
```
Comment (MarcusNotheis) [inline packages/main/src/webComponents/SelectMenuOption/index.tsx:49]: remove

File: packages/main/src/webComponents/SelectMenuOption/index.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+ *
+ * __Note:__ This component is a web component developed by the UI5 Web Components’ team.
+ *
+ * [UI5 Web Components Storybook](https://sap.github.io/ui5-webcomponents/playground/?path=/docs/main-SelectMenuOption)
+ */
+const SelectMenuOption = withWebComponent<SelectMenuOptionPropTypes, SelectMenuOptionDomRef>(
+  'ui5-select-menu-option',
+  ['displayText', 'type', 'value'],
+  ['navigated'],
+  ['deleteButton'],
```
Comment (MarcusNotheis) [inline packages/main/src/webComponents/SelectMenuOption/index.tsx:63]:
```suggestion
  ['displayText', 'value'],
  [],
  [],
```

File: packages/main/src/webComponents/SelectMenuOption/index.tsx [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+  [],
+  () => import('@ui5/webcomponents/dist/SelectMenuOption.js')
+);
+
+SelectMenuOption.displayName = 'SelectMenuOption';
+
+SelectMenuOption.defaultProps = {
+  type: ListItemType.Active
```
Comment (MarcusNotheis) [inline packages/main/src/webComponents/SelectMenuOption/index.tsx:71]: remove

File: packages/main/src/webComponents/Select/index.tsx [hunk 6]
```diff
@@ -23,6 +23,13 @@ interface SelectAttributes {
    * **Note:** A disabled component is noninteractive.
    */
   disabled?: boolean;
+  //todo adjust types and description + add story
+  /**
+   * Defines a reference (ID or DOM element) of component's menu of options. as alternative to define the select's dropdown.
+   *
+   * **Note:** Usage of `SelectMenu` is recommended.
+   */
+  menu?: string | HTMLElement;
```
Comment (MarcusNotheis) [inline packages/main/src/webComponents/Select/index.tsx:32]: should be only of type `string` here, the `string | HTMLElement` type should move to the `SelectDomRef`

**Review by MarcusNotheis — DISMISSED** [review]

File: packages/main/src/webComponents/SelectMenuOption/SelectMenuOptionDescription.md [hunk 7]
```diff
@@ -0,0 +1,3 @@
+## Usage
+
+For the `SelectMenuOption`
```
[the remark this answers is not in the document]
Comment (Lukas742) [reply packages/main/src/webComponents/SelectMenuOption/SelectMenuOptionDescription.md:3]: Nope, it's the actual [description](https://github.com/SAP/ui5-webcomponents/blob/d512116994e48998c2970d62180e22bbc249515d/packages/main/src/SelectMenuOption.ts#L23) :D 
ui5wc often use this, e.g. [here](https://github.com/SAP/ui5-webcomponents/blob/d512116994e48998c2970d62180e22bbc249515d/packages/fiori/src/SortItem.ts#L13)

File: packages/main/src/components/AnalyticalCardHeader/AnalyticalCardHeader.jss.ts [hunk 8]
```diff
@@ -15,16 +21,16 @@ const styles = {
...
-      right: 'var(--_ui5_card_header_focus_offset)',
-      bottom: 'var(--_ui5_card_header_focus_offset)',
-      borderTopLeftRadius: 'var(--_ui5_card_header_focus_radius)',
-      borderTopRightRadius: 'var(--_ui5_card_header_focus_radius)',
-      borderBottomLeftRadius: 'var(--_ui5_card_header_focus_bottom_radius)',
-      borderBottomRightRadius: 'var(--_ui5_card_header_focus_bottom_radius)'
+      top: _cardHeaderFocusOffset,
+      left: _cardHeaderFocusOffset,
+      right: _cardHeaderFocusOffset,
+      bottom: _cardHeaderFocusOffset,
```
Comment (MarcusNotheis) [inline packages/main/src/components/AnalyticalCardHeader/AnalyticalCardHeader.jss.ts:29]: I think we can replace these 4 properties with `inset: _cardHeaderFocusOffset`

**Review by MarcusNotheis — APPROVED** [review]

**Pull request merged**